lib/OpenQA/Events.pm

```diff
@@ -18,6 +18,20 @@ use Mojo::Base 'Mojo::EventEmitter';
 
 sub singleton { state $events = shift->SUPER::new }
 
+# emits an event via singleton
+# note: Supposed to be used from non-controller context. Use the equally named helper
+#       to emit events from a controller.
+sub emit_event {
+    my ($type, %args) = @_;
+    die 'missing event type' unless $type;
```

Comment:
Not a blocker, but probably worth mentioning (and fixing in a future PR). This should have been a class method, instead of a function.